### ISSUE TITLE
Issue when creating vnet with az cli

### ISSUE THREAD

**Issue by lorcopotia** [Azure/azure-cli#15984]
## Describe the bug

**Command Name**
`az network vnet create`

**Errors:**
```
'registrationState'
Traceback (most recent call last):
python3.6/site-packages/knack/cli.py, ln 215, in invoke
    cmd_result = self.invocation.execute(args)
cli/core/commands/__init__.py, ln 654, in execute
    raise ex
cli/core/commands/__init__.py, ln 718, in _run_jobs_serially
    results.append(self._run_job(expanded_arg, cmd_copy))
cli/core/commands/__init__.py, ln 711, in _run_job
    six.reraise(*sys.exc_info())
...
python3.6/site-packages/msrestazure/tools.py, ln 57, in register_rp_hook
    if not _register_rp(session, url_prefix, rp_name):
python3.6/site-packages/msrestazure/tools.py, ln 102, in _register_rp
    if rp_info['registrationState'] == 'Registered':
KeyError: 'registrationState'
```

## To Reproduce:
Steps to reproduce the behavior. Note that argument values have been redacted, as they may contain sensitive information.

- _Put any pre-requisite steps here..._
- `az network vnet create -g {} -n {} --address-prefixes {} --subnet-name {} --subnet-prefixes {}`

## Environment Summary
```
Linux-4.15.0-1098-azure-x86_64-with-debian-stretch-sid (Cloud Shell)
Python 3.6.10
Installer: DEB

azure-cli 2.14.0 *

Extensions:
ai-examples 0.2.4

```

**Comment by yonzhan** [comment]
network

**Comment by haroldrandom** [comment]
![image](https://user-images.githubusercontent.com/14357159/101442765-f37da300-3956-11eb-83a5-eadccc10fa27.png)

I cannot reproduce your situation. Could you try it again to see if it stably happened your environment ?

**Comment by lorcopotia** [comment]
Sorry I didn't reply sooner. I just executed the same command from the history and it worked. I don't know, I just logged back in, looked for the command and it worked.
I can see that there is another thing going on, but everything is fine with me now.
FYI the commands are:

`$rg = (Get-AzResourceGroup).ResourceGroupName` 
`az network vnet create -g $rg -n CloudLorcoVNet --address-prefixes 10.15.0.0/16 --subnet-name WebSubnet --subnet-prefixes 10.15.0.0/24`

Thanks!

> ![image](https://user-images.githubusercontent.com/14357159/101442765-f37da300-3956-11eb-83a5-eadccc10fa27.png)
> 
> I cannot reproduce your situation. Could you try it again to see if it stably happened your environment ?
